ie_serving/server/service.py

```diff
@@ -80,49 +85,25 @@ def Predict(self, request, context):
             logger.debug("PREDICT, problem with input data. Exit code {}"
                          .format(code))
             return predict_pb2.PredictResponse()
+
         target_engine = self.models[model_name].engines[version]
-        target_engine.in_use.acquire()
-        ################################################
-        # Reshape network inputs if needed
-        reshape_param = target_engine.detect_shapes_incompatibility(
-            inference_input)
-        if reshape_param is not None:
-            error_message = target_engine.reshape(reshape_param)
-            if error_message is not None:
-                code = statusCodes['invalid_arg'][GRPC]
-                context.set_code(code)
-                context.set_details(error_message)
-                target_engine.in_use.release()
-                return predict_pb2.PredictResponse()
-        ################################################
-        inference_start_time = datetime.datetime.now()
-        inference_output, error_message = target_engine.infer(
-            inference_input)
-        if error_message is not None:
+        inference_id = str(uuid.uuid1())

```

Comment:
How about having a pool of IDs? 
If we specify max number of workers for the server, then we know that the number of requests handled in parallel would not exceed number of workers. We could generate that pools on OVMS start-up and then every worker that starts processing request would take free ID from the pool and give it back after it finishes instead of generating new ID per request. 
What do you think? Would it be safe or am I missing something?